lib/solarwinds_apm/support/resource_detector/aws/eks.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+# © 2023 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+require 'net/http'
+require 'uri'
+require 'json'
+require 'socket'
+
+module SolarWindsAPM
+  module ResourceDetector
+    module EKS
```

Comment:
Curious if this is inspired by some upstream detector since it also seems to use the Kubernetes API, which would seem to present quite a set up hurdle.

Reply:
I took [otel js](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/detectors/node/opentelemetry-resource-detector-aws/src/detectors) as the reference

Reply:
i see, seems JS is based on the original [Java](https://github.com/open-telemetry/opentelemetry-java/pull/1671) implementation and requires the CW monitoring enabled in order to get cluster name, but the container ID detection should work and could be of interest to us.